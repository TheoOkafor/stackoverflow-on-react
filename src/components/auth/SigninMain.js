import React, { Component } from 'react';
import SigninForm from './SigninForm';

class SigninMain extends Component {
  render() {
    return (
      <div className="main row-container">
        <div className="sd-margin-12 d-margin-6 margin-top-20 site-desc">
          <h1 className="heading center">
            Welcome back to Stack Overflow<small>-lite</small>
          </h1>
          <p>
            This is a platform where you can ask
            <span>questions</span> and provide <span>answers</span>.
          </p>
        </div>

        <SigninForm />
        <div className="clear-fix" />

      </div>
    );
  }
}

export default SigninMain;
