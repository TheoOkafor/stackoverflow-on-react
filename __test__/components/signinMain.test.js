import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import SigninMain from '../../src/components/auth/SigninMain';
import store from '../../src/store';

test('Test that the Signup Main component displays', () => {
  const component = renderer.create(
    <Provider store={store}>
      <SigninMain />
    </Provider>
  );
  expect(component).toMatchSnapshot();
});
