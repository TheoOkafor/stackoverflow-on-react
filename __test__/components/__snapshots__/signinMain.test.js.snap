// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the Signup Main component displays 1`] = `
<div
  className="main row-container"
>
  <div
    className="sd-margin-12 d-margin-6 margin-top-20 site-desc"
  >
    <h1
      className="heading center"
    >
      Welcome back to Stack Overflow
      <small>
        -lite
      </small>
    </h1>
    <p>
      This is a platform where you can ask
      <span>
        questions
      </span>
       and provide 
      <span>
        answers
      </span>
      .
    </p>
  </div>
  <div
    className="sd-margin-12 d-margin-6"
  >
    <div
      className="sign-card"
    >
      <form
        className="signup-form"
        onSubmit={[Function]}
      >
        <p>
          <small>
            Don't have an account?
          </small>
          <a
            className="inherit"
            href="/signup"
          >
            <b>
               Sign up
            </b>
          </a>
        </p>
        <p
          className="signup-error "
        >
          
        </p>
        <div
          className="form-item"
        >
          <label
            htmlFor="email"
          >
            Email
          </label>
          <input
            data-testid="signin-email"
            name="email"
            onInput={[Function]}
            placeholder="Email"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-item"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            data-testid="signin-password"
            id="password"
            name="password"
            onInput={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
        </div>
        <button
          className="btn primary"
          data-testid="signinBtn"
          id="sign-in"
        >
          Sign in
        </button>
      </form>
    </div>
  </div>
  <div
    className="clear-fix"
  />
</div>
`;
